ern/index.go

```diff
@@ -268,55 +273,153 @@ func (i *Indexer) indexResourceList(ernID *cid.Cid, obj *meta.Object) error {
 func (i *Indexer) indexSoundRecording(ernID *cid.Cid, obj *meta.Object) error {
 	graph := meta.NewGraph(i.store, obj)
 
-	// load each potential ID separately
-	var ids []string
-	for _, field := range []string{"ISRC", "CatalogNumber", "ProprietaryId"} {
-		v, err := graph.Get("SoundRecordingId", field, "@value")
-		if meta.IsPathNotFound(err) {
-			continue
-		} else if err != nil {
+	// Only *attempt* to load the ISRC, other IDs can be retrieved via GraphQL
+	// Default to empty string if not present
+	var isrc string
+	v, err := graph.Get("SoundRecordingId", "ISRC", "@value")
+	if err == nil {
+		isrc = v.(string)
+	}
+	
+
+	// Insert the DisplayArtist to party table
+	srCid, err := obj.Get("SoundRecordingDetailsByTerritory")
+	if err != nil {
+		return err
+	}
+	var cids []*cid.Cid
+	switch srCid := srCid.(type) {
+	case *format.Link:
+		cids = []*cid.Cid{srCid.Cid}
+	case []interface{}:
+		for _, x := range srCid {
+			cid, ok := x.(*cid.Cid)
+			if !ok {
+				return fmt.Errorf("invalid resource type %T, expected *cid.Cid", x)
+			}
+			cids = append(cids, cid)
+		}
+	}
+	for _, cid := range cids {
+		obj, err := i.store.Get(cid)
+		if err != nil {
+			return err
+		}
+		id, err := InsertParty(i, obj, "DisplayArtist")
+		if err != nil {
 			return err
 		}
-		ids = append(ids, v.(string))
+		_ = id
```

Comment:
If we don;t care about the returned `id` then assign it to `_` above:

```go
_, err := i.insertParty(...)
```

Reply:
Thanks, neater.

Reply:
Neater, thanks.